bms_ranges_test.py

```diff
@@ -0,0 +1,19 @@
+import unittest
+import bms_ranges_monitor
+
```

Comment:
In TDD, we write a test and then write the code to satisfy the test.
Here, the code is doing more than that, it's printing the ranges.
The crucial part of the functionality - the ranges - is absent from the tests. That means you didn't use the tests to derive the structure of the output data.
Separate the functionality into a pure function and test its return value (e.g., as a dictionary). Keep it separate from the printing functionality. Write the test for checking the return value of this pure function and then implement.